Getting started improvements (#179)
TheThingsNetwork/lorawan-stack: doc/gettingstarted.md

## Dependencies 

### CLI and Stack

> For **stable** versions, see the [Releases](https://github.com/TheThingsNetwork/lorawan-stack/releases) page on Github.

Reviewer: ```suggestion
Get the latest packages or binaries for your operating system from the [releases](https://github.com/TheThingsNetwork/lorawan-stack/releases) page on GitHub.
```

Reviewer: How about mentioning common commands for Mac (`brew ...`) and Ubuntu (`apt ...`)?

Reviewer 2: I think it would be appropriate to put those instructions in the descriptions of the releases.

Reviewer: I'm not so sure. It would be in every release, and it's what people do but it misses the context of getting started. That's here.

Also, I'd like to avoid that people have to go to other pages in order to pass the getting started. A self-containing getting started would be ideal for me.

Reviewer 2: But then where does it end? Is the Getting Started going to tell how to install the binaries using Apt? Brew? Yum? Nix? Chocolatey? Which files to download? IMO that belongs on a "Downloads" page that the Getting Started will refer to.

If we combine that with the Releases, we can point people to the exact binary versions instead of having to specify that in some other way.

Reviewer: Yeah it's a sliding edge, I agree. OK fine to put it in the releases and link there then.

Reviewer: @adriansmares so my suggestion here as first comment is still valid

### Certificates

By default, the Stack requires a `cert.pem` and `key.pem`, in order to to serve content over TLS.

+ To generate self-signed certificates, you can use the following command. This requires a [Go environment setup](../DEVELOPMENT.md#development-environment).

```bash

> go run $(go env GOROOT)/src/crypto/tls/generate_cert.go -ca -host localhost && chmod 0444 ./key.pem

Reviewer: Note that `chmod` doesn't work on Windows

Author: All of the commands are marked as `bash`, and Windows isn't a first class citizen anyway (due to the Makefiles).

```
    

> + Keep in mind that self-signed certificates are not trusted by browsers and operating systems, and as such they will return a warning regarding this matter. In order to avoid such issues, we recommend [LetsEncrypt](https://letsencrypt.org/getting-started/).

Reviewer: ```suggestion
+ Keep in mind that self-signed certificates are not trusted by browsers and operating systems, and as such they will return a warning regarding this matter. In order to avoid such issues, we recommend [Let's Encrypt](https://letsencrypt.org/getting-started/).
```

<a name="configuration"/>

> ## Configuration 

Reviewer: Does `## <a name="configuration">Configuration</a>` not work?

Author: The rendering no longer works correctly when Markdown is combined with HTML.

## <a name="configuration">Configuration</a>

By default, the stack allows unregistered gateways to connect, but without providing a default band. As such, it is highly recommended that each gateway is registered:

```bash

> $ ttn-lw-cli gateway create gtw1 --user-id admin --frequency_plan_id EU_863_870 --gateway-eui 00800000A00009EF

Reviewer: ```suggestion
$ ttn-lw-cli gateway create gtw1 --user-id admin --frequency_plan_id EU_863_870 --gateway-eui 00800000A00009EF --enforce-duty-cycle
```

We should probably introduce a defaults flags here too.

Reviewer 2: I agree, but that deserves its own issue

Reviewer: For sure

```

> This creates a gateway `gtw1` with the frequency plan `EU_863_870` and EUI `00800000A00009EF`. You can now connect 

Reviewer: ```suggestion
This creates a gateway `gtw1` with the frequency plan `EU_863_870` and EUI `00800000A00009EF` that respects duty-cycle limitations. For more options, you can use the `--help` flag. 
```

> your gateway to the stack. For more details you can use the `--help` flag.

Reviewer: ```suggestion
You can now connect your gateway to the stack.
```

You can now register an OTAA activated device to be used with the stack as follows:

```bash

> $ ttn-lw-cli end-devices create --device-id dev1 --dev-eui 0004A30B001C0530 --join-eui 800000000000000C --application-id app1 --frequency_plan_id EU_863_870 --root_keys.app_key.key 752BAEC23EAE7964AF27C325F4C23C9A --lorawan_phy_version 1.0.2-b --lorawan_version 1.0.2

Reviewer: ```suggestion
$ ttn-lw-cli devices create app1 dev1 --dev-eui 0004A30B001C0530 --join-eui 800000000000000C --frequency_plan_id EU_863_870 --root_keys.app_key.key 752BAEC23EAE7964AF27C325F4C23C9A --lorawan_phy_version 1.0.2-b --lorawan_version 1.0.2
```

Reviewer 2: `end-devices` was fine, this is the official term, and also how they're called in our API (except where #170 still needs to fix it). `device`, `dev`, `ed` and `d` are simply aliases for `end-devices`.

Reviewer: OK for the official term. I'm aware of aliases.
